x-pack/plugins/apm/scripts/create-functional-tests-archive/index.ts

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { argv } from 'yargs';
+import { execSync } from 'child_process';
+import moment from 'moment';
+import path from 'path';
+import fs from 'fs';
+import { stampLogger } from '../shared/stamp-logger';
+
+async function run() {
+  stampLogger();
+  const archiveName = (argv.name as string | undefined) ?? 'apm_8.0.0';
```

Comment:
Instead of testing against 8.0.0 (aka master), should we start testing against released versions?
Since 8.0.0 is not as constant as the version implies, an archive made in the very beginning of the 8.0 development will be different from an archive made when 8.0.0 is actually released.

We are also missing out on the opportunity to have the same test run against several different versions (to ensure backwards-compatability). 
We might not want snapshots for every minor but having a couple per major would be nice.

Instead of letting the user specify the archive name, I think it should be set automatically based on the stack version by calling `GET ${esUrl}`

Reply:
> Instead of letting the user specify the archive name, I think it should be set automatically based on the stack version by calling GET ${esUrl}

Correction: instead of retrieving the version from ES we should get it from the APM Server that ingested the docs. We can get that by retrieving a sample doc:

![image](https://user-images.githubusercontent.com/209966/92475573-e88f6700-f1dd-11ea-9a90-a4bd65040cbb.png)

Reply:
I think locking it down like this would be the real benefit of using this script: if we have a strict process for updating the archives that's encoded in code it's impossible to screw it up. Just run the script, commit the new archives and call it a day.

Reply:
I would be okay with locking it down further as a start. @cauemarcondes do you think there's still a need for separate archives (if so, could you clarify what should be configurable)?

I'm not sure if we should worry _too_ much about versions. I imagine this snapshot would be regularly updated, and in general code from earlier versions works with newer versions (across a major). WDYT about seeing how a simple approach works? We can always improve later.

Reply:
>  I imagine this snapshot would be regularly updated and in general code from earlier versions works with newer versions (across a major). 

Isn't that the problem though? that we always code we the expectation that a certain field is always available, and forget to test older versions where it's not?

> WDYT about seeing how a simple approach works? We can always improve later.

I'm good with the simple solution 👍